src/client/cssom/computed_style.cpp

```diff
@@ -182,25 +183,61 @@ namespace client_cssom
     if (from_style.length() == 0)
       return false;
 
+    // Two-pass approach: first process custom properties, then regular properties
+    // This ensures custom properties are available when resolving variables in regular properties
+
+    // First pass: Process all custom properties (CSS variables)
     for (int index = 0; index < from_style.length(); index++)
     {
       const auto propertyName = from_style.item(index);
-      const auto value = from_style.getPropertyValue(propertyName);
 
-      auto it = find(propertyName);
-      if (it != end())
+      // Only process custom properties in the first pass
+      if (propertyName.length() >= 2 && propertyName.substr(0, 2) == "--")
       {
-        if (it->second == value)
-          continue;         // No need to update if the value is the same.
-        it->second = value; // Update the existing property.
+        const auto value = from_style.getPropertyValue(propertyName);
+
+        auto it = find(propertyName);
+        if (it != end())
+        {
+          if (it->second == value)
+            continue;         // No need to update if the value is the same.
+          it->second = value; // Update the existing property.
+        }
+        else
+        {
+          setPropertyInternal(propertyName, value);
+        }
+
+        if (context.has_value())
+          computeProperty(propertyName, value, context.value());
```

Comment:
No need to call `computeProperty()` at first pass that updates the custom properties only.